gammapy/detect/tests/test_test_statistics.py

```diff
@@ -38,3 +39,11 @@ def test_compute_ts_map():
     assert_allclose([[99], [99]], np.where(result.ts == result.ts.max()))
     assert_allclose(6, result.niter[99, 99])
     assert_allclose(1.0227934338735763e-09, result.amplitude[99, 99], rtol=1e-3)
+
+    # test write method
+    with NamedTemporaryFile(suffix='.fits') as f:
```

Comment:
I think the recommended way to use file objects is the `with` context. I don't know about temporary files, but for 'normal' files this ensures, that the file is closed again, even if the code fails in between. And `f.close()`, which does the garbage collection and maybe deletes the temporary file, doesn't have to be called explicitly. In some places in Gammapy, where `NamedTemporaryFile` objects are used, they aren't closed again, which I think is bad style and could lead to unexpected errors.